src/math/Math.sol

```diff
@@ -28,4 +28,16 @@ library Math {
             }
         }
     }
+
+    /// @dev Returns x / y rounded up.
+    function divUp(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        assembly {
+            // Revert if a + b < a or a + b < b
+            z := add(x, y)
+            if or(iszero(y), or(lt(z, y), lt(z, x))) {
+                revert(0, 0)
+            }
+            z := div(sub(z, 1), y)
```

Comment:
Save 9 gas.
(Need to define at the beginning: `uint256 internal constant MAX_UINT256 = 2**256 - 1;`)

```suggestion
            z := add(x, y)
            if or(iszero(y), or(lt(z, y), lt(z, x))) {
                revert(0, 0)
            }
            z := div(sub(z, 1), y)
```


```suggestion
            if or(iszero(y), gt(x, sub(MAX_UINT256, y))) {
                revert(0, 0)
            }
            z := div(sub(add(x, y), 1), y)
```
![image](https://user-images.githubusercontent.com/87616492/181578325-daddf454-eb0d-4218-82be-fc83ac4fcdd0.png)

Reply:
```solidity
assembly {
            z := add(x, y)
            if or(iszero(y), lt(z, x)) {
                revert(0, 0)
            }
            z := div(sub(z, 1), y)
}```

This version is even better -> 448